### PR TITLE
Fixes #68: Beta provider support

### LINKED ISSUE #68: Beta provider support
Do we intend to have this module expose beta options as a separate module similar to how the GKE module provides this by using templating?

In my use case next hop ILB for routes is currently in the [beta provider](https://www.terraform.io/docs/providers/google/r/compute_route.html#next_hop_ilb) and i would like to see it exposed in this module.

That being said there is a workaround (declare beta route resource outside this module), so it's not a blocker but more of a nice to have.

### LINKED ISSUE #114: Add `next_hop_ilb`
Add `next_hop_ilb` for `google_compute_route` (beta). This will come after support for the beta provider has been added #68

### PR DESCRIPTION
**Pull request by nick4fake**
Fixes https://github.com/terraform-google-modules/terraform-google-network/issues/68
Fixes #114

Added beta submodule.

### CODE REVIEW

**Comment by nick4fake** [comment]
@morgante, @aaron-lane linter fails due to codelabs example

File: examples/ilb_routing/versions.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = "~> 0.12.0"
```
Comment (aaron-lane) [inline examples/ilb_routing/versions.tf]: We should ensure compatibility with `for_each`.

```suggestion
  required_version = "~> 0.12.6"
```

File: modules/routes-beta/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+-  google_compute_route.next_hop_ilb
+
+## Usage
+
+Basic usage of this submodule is as follows:
+
+```hcl
+module "vpc" {
+    source  = "terraform-google-modules/network/google//modules/routes"
```
Comment (aaron-lane) [inline modules/routes-beta/README.md]:
```suggestion
    source  = "terraform-google-modules/network/google//modules/routes-beta"
```

File: modules/routes-beta/versions.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = "~> 0.12.0"
```
Comment (aaron-lane) [inline modules/routes-beta/versions.tf]:
```suggestion
  required_version = "~> 0.12.6"
```

File: modules/subnets-beta/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+-  google_compute_subnetwork.purpose
+-  google_compute_subnetwork.role
+
+## Usage
+
+Basic usage of this submodule is as follows:
+
+```hcl
+module "vpc" {
+    source  = "terraform-google-modules/network/google//modules/subnets"
```
Comment (aaron-lane) [inline modules/subnets-beta/README.md]:
```suggestion
    source  = "terraform-google-modules/network/google//modules/subnets-beta"
```

File: modules/routes-beta/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+It also uses google beta provider to support the following resource fields:
+
+-  google_compute_route.next_hop_ilb
+
+## Usage
+
+Basic usage of this submodule is as follows:
+
+```hcl
+module "vpc" {
```
Comment (aaron-lane) [inline modules/routes-beta/README.md]: This example should include `next_hop_ilb`.

File: modules/subnets-beta/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+-  google_compute_subnetwork.purpose
+-  google_compute_subnetwork.role
+
+## Usage
+
+Basic usage of this submodule is as follows:
+
+```hcl
+module "vpc" {
```
Comment (aaron-lane) [inline modules/subnets-beta/README.md]: This example should include `purpose` and `role`.

File: modules/subnets-beta/versions.tf [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = "~> 0.12.0"
```
Comment (aaron-lane) [inline modules/subnets-beta/versions.tf]:
```suggestion
  required_version = "~> 0.12.6"
```

**Pull request merged**